app/controllers/items/new.js

```diff
@@ -463,13 +504,29 @@ export default GoodcityController.extend(
             this.printBarcode(data.item.id);
           }
           this.updateStoreAndSaveImage(data);
-          this.clearSubformAttributes();
-          this.setProperties({
-            locationId: "",
-            inventoryNumber: "",
-            offersLists: []
-          });
-          this.replaceRoute("items.detail", data.item.id);
+          if (this.get("isDuplicate") && !this.get("isBoxOrPallet")) {
+            this.replaceRoute("items.new");
+            this.paramsToBeCleared();
+            this.set("quantity", 1);
+            this.send("autoGenerateInventoryNumber");
```

Comment:
I'm guessing this method returns a promise (please double check), if so you likely want to await that method

Reply:
yes added await for the required methods